core/src/test/java/org/apache/calcite/rex/RexSimplifierTest.java

```diff
@@ -0,0 +1,525 @@
+/*
```

Comment:
If possible, it would be cool to integrate with existing tests in RexProgramTest instead of adding new infra again, as they already contain multiple tests for CASE (and other preds) simplification. 

https://github.com/apache/calcite/blob/master/core/src/test/java/org/apache/calcite/test/RexProgramTest.java#L1101